src/zenml/config/build_configuration.py

```diff
@@ -60,7 +60,7 @@ def compute_settings_checksum(
             The checksum.
         """
         hash_ = hashlib.md5()  # nosec
-        hash_.update(self.settings.json().encode())
+        hash_.update(self.settings.model_dump_json().encode())
```

Comment:
I feel that `.json()` was returning a `string`, while `.model_dump_json()` returns `Dict[str,Any]`. Is it wrong?

Reply:
Ah, my bad it is `model_json_schema` returning Dict